Examples/test-suite/go/li_constraints_runme.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import wrap "swigtests/li_constraints"
+
+func check_double(except bool, f func (float64), val float64) {
+    actual := true;
+	func() {
+		defer func() {
+			if recover() != nil {
+                actual = false
+			}
+		}()
+        f(val);
+	}()
+    if actual != except {
+      if actual {
+          panic("function did not perform exception")
+      } else {
+          panic("function should perform exception")
+      }
+    }
+}
+
+func main() {
+    check_double(true, wrap.Test_nonnegative, 10);
+    check_double(true, wrap.Test_nonnegative, 0);
+    check_double(false, wrap.Test_nonnegative, -10);
+
+    check_double(false, wrap.Test_nonpositive, 10);
+    check_double(true, wrap.Test_nonpositive, 0);
+    check_double(true, wrap.Test_nonpositive, -10);
+
+    check_double(true, wrap.Test_positive, 10);
+    check_double(false, wrap.Test_positive, 0);
+    check_double(false, wrap.Test_positive, -10);
+
+    check_double(false, wrap.Test_negative, 10);
+    check_double(false, wrap.Test_negative, 0);
+    check_double(true, wrap.Test_negative, -10);
+
+    check_double(true, wrap.Test_nonzero, 10);
+    check_double(false, wrap.Test_nonzero, 0);
+    check_double(true, wrap.Test_nonzero, -10);
+
+    have_exception := false
+	func() {
+		defer func() {
+			if recover() != nil {
+                have_exception = true
+			}
+		}()
+        // The NULL value
+        wrap.Test_nonnull(uintptr(0));
+	}()
+    if !have_exception {
+        panic("test_nonnull should perform exception with 'null' value")
+    }
+    /* The pointer value is not actually used,
+     * We just want it to be a non-null */
+    wrap.Test_nonnull(uintptr(111));
```

Comment:
Does this end up doing something like `(void*)111`?  If so, that's undefined behaviour isn't it?

I'd suggest adding a helper method to `li_constraints.i` which returns a valid non-NULL `void*` pointer, e.g.:

````
void* get_nonnull() {
    static int i;
    return &i;
}
```

That should then allow testing this for all the target languages, e.g. here it would be something like:

```
wrap.Test_nonnull(wrap.Get_nonnull());
```

Reply:
Sound a good continue.
I was only seeking to write as many tests as I can without change li_constraints.i.
Only bug fixes.
But your idea should work with most other languages, so it will actually complete the test with other languages.
Only go language can cast to void.
Yes `uintptr(111)` equal `(void*)111`. 
As all pointers from go are translate to `uintptr` in our wrapper.

Reply:
Probably new commits is simpler for review.  I can squash as I merge if appropriate.